enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2029,6 +2034,14 @@ func (e *partitionEvictor) sampleGroupID() (string, error) {
 		return "", err
 	}
 
+	// The special ANON group doesn't follow the regular group key format, so it
+	// won't be found through regular sampling. To get around this, we make up
+	// a random group ID for it and pretend it exists in the keyspace so that
+	// it can be sampled like any other group.
+	if randomGroupID < e.anonPseudoGroupID && (key.GroupID() == "" || key.GroupID() > e.anonPseudoGroupID) {
```

Comment:
could you add a test for this functionality?

Reply:
The test in https://github.com/buildbuddy-io/buildbuddy/pull/4091/files already uses anon data. You could make an argument that we should be including non-anon data there as well.